spring-restdocs-core/src/main/java/org/springframework/restdocs/ManualRestDocumentation.java

```diff
@@ -31,17 +33,31 @@
  */
 public final class ManualRestDocumentation implements RestDocumentationContextProvider {
 
+	private static final String GENERATED_SNIPPETS_PATH = "generated-snippets";
+	private static final String MAVEN_TARGET_PATH = "target" + File.separator + GENERATED_SNIPPETS_PATH;
+	private static final String GRADLE_BUILD_PATH = "build" + File.separator + GENERATED_SNIPPETS_PATH;
+	private static final String MAVEN_POM = "pom.xml";
+	private static final String OUTPUT_DIR_PROPERTY_KEY = "snippetOutputDirectory";
+
 	private final File outputDirectory;
 
 	private RestDocumentationContext context;
 
 	/**
 	 * Creates a new {@code ManualRestDocumentation} instance that will generate snippets
+	 * to &lt;gradle/maven build path&gt;/generated-snippet if no runtime property ({@code snippetOutputDirectory}) is set.
+	 */
+	public ManualRestDocumentation() {
+		this(System.getProperty(OUTPUT_DIR_PROPERTY_KEY, getDefaultOutputDirectory()));
+	}
+
+	/**
+	 * Creates a new {@code ManualRestDocumentation} instance that will generate snippets
 	 * to the given {@code outputDirectory}.
 	 *
 	 * @param outputDirectory the output directory
 	 */
-	public ManualRestDocumentation(String outputDirectory) {
```

Comment:
This constructor needs to remain public, both to allow the output directory to be overridden and to maintain backwards compatibility.